src/Cafe/OS/libs/dmae/dmae.cpp

```diff
@@ -36,6 +36,16 @@ void dmaeExport_DMAECopyMem(PPCInterpreter_t* hCPU)
 			dstBuffer[i] = _swapEndianU32(srcBuffer[i]);
 		}
 	}
+	else if( hCPU->gpr[6] == DMAE_ENDIAN_16 )
+	{
+		// swap per uint16
+		uint16* srcBuffer = (uint16*)memory_getPointerFromVirtualOffset(hCPU->gpr[4]);
+		uint16* dstBuffer = (uint16*)memory_getPointerFromVirtualOffset(hCPU->gpr[3]);
+		for(uint32 i=0; i<hCPU->gpr[5]; i++)
```

Comment:
I am unsure but it might be size * 2 here because it counts in multiple of 4 bytes. Do you know what behavior the homebrew expects?

Reply:
You are definitely right, I didn't take into account the size * 2 in the homebrew code and it ended up working anyway...